Source/Meadow.Units/PotentialHydrogen.cs

```diff
@@ -0,0 +1,358 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.Contracts;
+using System.Runtime.InteropServices;
+
+namespace Meadow.Units
+{
+    /// <summary>
+    /// Represents potential hydrogen (ph)
+    /// </summary>
+    [Serializable]
+    [ImmutableObject(true)]
+    [StructLayout(LayoutKind.Sequential)]
+    public struct PotentialHydrogen :
+        IComparable, IFormattable, IConvertible,
+        IEquatable<double>, IComparable<double>
+    {
+        /// <summary>
+        /// Creates a new `PotentialHydrogen` object.
+        /// </summary>
+        /// <param name="value">The Potential Hydrogen value.</param>
+        /// <param name="type">Potential Hydrogen unit.</param>
+        public PotentialHydrogen(double value, UnitType type = UnitType.pH)
+        {
+            pH = value;
+        }
+
+        /// <summary>
+        /// Creates a new `PotentialHydrogen` object from an existing PotentialHydrogen object
+        /// </summary>
+        /// <param name="PotentialHydrogen"></param>
+        public PotentialHydrogen(PotentialHydrogen PotentialHydrogen)
+        {
+            pH = PotentialHydrogen.pH;
+        }
+
+        /// <summary>
+        /// The Potential Hydrogen expressed as pH.
+        /// </summary>
+        public double pH { get; private set; }
+        /// <summary>
+        /// The type of units available to describe the temperature.
+        /// </summary>
+        public enum UnitType
+        {
+            /// <summary>
+            /// Potential Hydrogen (pH)
+            /// </summary>
+            pH
+        }
+
+        /// <summary>
+        /// Compare to another PotentialHydrogen object
+        /// </summary>
+        /// <param name="obj">The object to compare</param>
+        /// <returns>true if equal</returns>
+        [Pure]
+        public override bool Equals(object obj)
+        {
+            if (obj is null) { return false; }
+            if (Equals(this, obj)) { return true; }
+            return obj.GetType() == GetType() && Equals((PotentialHydrogen)obj);
+        }
+
+        /// <summary>
+        /// Get hash of object
+        /// </summary>
+        /// <returns>int32 hash value</returns>
+        [Pure] public override int GetHashCode() => pH.GetHashCode();
+
+        // implicit conversions
+        //[Pure] public static implicit operator PotentialHydrogen(ushort value) => new PotentialHydrogen(value);
```

Comment:
Are these meant to be included or deleted?

Reply:
copy/paste laziness ..... we have them commented out in all of the units
 
I'll file an issue to investigate